src/Sylius/Bundle/AdminBundle/Form/Extension/CatalogPromotionScope/ForProductsScopeConfigurationTypeExtension.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Form\Extension\CatalogPromotionScope;
+
+use Sylius\Bundle\AdminBundle\Form\Type\ProductAutocompleteChoiceType;
+use Sylius\Bundle\CoreBundle\Form\Type\CatalogPromotionScope\ForProductsScopeConfigurationType;
+use Sylius\Component\Core\Model\ProductInterface;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\FormBuilderInterface;
+
+final class ForProductsScopeConfigurationTypeExtension extends AbstractTypeExtension
+{
+    /** @param DataTransformerInterface<ProductInterface, string|null> $productsToCodesTransformer */
+    public function __construct(private readonly DataTransformerInterface $productsToCodesTransformer)
+    {
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder->add('products', ProductAutocompleteChoiceType::class, [
+            'label' => 'sylius.ui.products',
+            'multiple' => true,
+            'required' => false,
+            ])
```

Comment:
```suggestion
        $builder->add('products', ProductAutocompleteChoiceType::class, [
                'label' => 'sylius.ui.products',
                'multiple' => true,
                'required' => false,
            ])
```
and in below extensions